lecturework.html

```diff
@@ -60,52 +60,54 @@ <h3>
     </div>
 
     <script>
-
+      const shoppingCart = [];
       const phoneToCartEl = document.getElementById("addPhoneToCart");
       const nintendoToCartEl = document.getElementById("addNintendoToCart");
       const macbookToCartEl = document.getElementById("addMacbookToCart");
       const showShoppingCartEl = document.getElementById("showShoppingCart");
-      const totalPriceEl = document.getElementById("total");
+      const totalPriceEl = document.getElementById("totalPrice");
 
       function addIphoneToShoppingCart() {
         const product = {
           name: "Iphone",
           price: 900,
       }
-      shoppingCart.push(product)
+      shoppingCart.push(product);
       }
 
       function addNintendoToShoppingCart() {
         const product = {
           name: "Nintendo DS", 
           price: 1200,
       }
-      shoppingCart.push(product)
+      shoppingCart.push(product);
       }
 
       function addMacbookToShoppingCart() {
         const product = {
           name: "Mac Book Air",
           price: 3200,
         }
-        shoppingCart.push(product)
+        shoppingCart.push(product);
       };
       function showShoppingCartProducts() {
-        const shoppingCartContainer = document.getElementById("shoppingCart");
         for (let i=0; i < shoppingCart.length; i++) {
+          const shoppingCartContainer = document.getElementById("shoppingCart");
           const productContainer = document.createElement("div");
           productContainer.innerHTML = `
           ${shoppingCart[i].name}<br>
           ${shoppingCart[i].price}
           `;
-          shoppingCartContainer.appendChild(productContainer)
+          shoppingCartContainer.appendChild(productContainer);
+          shoppingCartContainer.innerHTML = ""
```

Comment:
Here is the issue of your code.
this is cleaning your  shoppingCartContainer each time that the for loop is running.
You have to do this just once before the for loop .

Reply:
so it's about 'where' i'm going to place this line:
shoppingCartContainer.innerHTML = "";

well, i'm not sure where to place it, but if it needs to go 'before' the for loop, then i place it here:

function showShoppingCartProducts() {
        shoppingCartContainer.innerHTML = ""
        for (let i=0; i < shoppingCart.length; i++) {
          const shoppingCartContainer = document.getElementById("shoppingCart");
          const productContainer = document.createElement("div");
          productContainer.innerHTML = `
          ${shoppingCart[i].name}<br>
          ${shoppingCart[i].price}
          `;
          shoppingCartContainer.appendChild(productContainer);
        }
      };